nifi-registry/nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/service/RegistryService.java

```diff
@@ -539,19 +539,23 @@ public VersionedFlow deleteFlow(final String bucketIdentifier, final String flow
     // ---------------------- VersionedFlowSnapshot methods ---------------------------------------------
 
     public VersionedFlowSnapshot createFlowSnapshot(final VersionedFlowSnapshot flowSnapshot) {
+        return createFlowSnapshot(flowSnapshot,false);
+    }
+    public VersionedFlowSnapshot createFlowSnapshot(final VersionedFlowSnapshot flowSnapshot, final boolean preserveSourceProperties) {
         if (flowSnapshot == null) {
             throw new IllegalArgumentException("Versioned flow snapshot cannot be null");
         }
 
         // validation will ensure that the metadata and contents are not null
-        if (flowSnapshot.getSnapshotMetadata() != null) {
+        if (flowSnapshot.getSnapshotMetadata() != null && !preserveSourceProperties ) {
```

Comment:
Similar to the approach in #7604, it looks like this method could be modified to check for a timestamp value already specified in the calling method. That would avoid having to pass down the `preserveSourceProperties` argument to this level.

Reply:
> Similar to the approach in #7604, it looks like this method could be modified to check for a timestamp value already specified in the calling method. That would avoid having to pass down the `preserveSourceProperties` argument to this level.

Hi @exceptionfactory  
I have made the requested change please have a look